backend/src/routes/aggregators/utils.ts

```diff
@@ -0,0 +1,516 @@
+    resource.spec.destination?.server === 'https://kubernetes.default.svc'
+  ) {
+    return 'local-cluster'
+  } else {
+    return getArgoDestinationCluster(resource.spec.destination, clusters, resource.status.cluster)
+  }
+}
+
+export function getArgoDestinationCluster(
+  destination: { name?: string; namespace: string; server?: string },
+  clusters: Cluster[],
+  cluster?: string
+) {
+  // cluster is the name of the managed cluster where the Argo app is defined
+  let clusterName = ''
+  const serverApi = destination?.server
+  if (serverApi) {
+    if (serverApi === 'https://kubernetes.default.svc') {
+      clusterName = cluster ?? 'Local'
+    } else {
+      const server = clusters.find((cls) => cls.kubeApiServer === serverApi)
+      clusterName = server ? server.name : 'unknown'
+    }
+  } else {
+    // target destination was set using the name property
+    clusterName = destination?.name || 'unknown'
+    if (cluster && (clusterName === 'in-cluster' || clusterName === 'local-cluster')) {
+      clusterName = cluster
+    }
+
+    if (clusterName === 'in-cluster') {
+      clusterName = 'local-cluster'
+    }
+  }
+  return clusterName
+}
+
+//////////////////////////////////////////////////////////////////
+// /////////////////// map created from events.ts /////////////////
+//////////////////////////////////////////////////////////////////
+export type ClusterMapType = {
+  [key: string]: IResource
+}
+export function getClusterMap(): ClusterMapType {
+  const managedClusters = getKubeResources('ManagedCluster', 'cluster.open-cluster-management.io/v1')
+  return managedClusters.reduce((clusterMap, cluster) => {
+    if (cluster.metadata.name) {
+      clusterMap[cluster.metadata.name] = cluster
+    }
+    return clusterMap
+  }, {} as ClusterMapType)
+}
+
+/////////////////////////////////////////////////////////////////////////////////
+// ///////// DISTRIBUTE APP QUERIES OVER MULTIPLE SEARCHES /////////////////////
+////////////////////////////////////////////////////////////////////////////////
+export type ApplicationPageChunk = {
+  keys?: string[]
+  limit: number
+}
+
+export function getNextApplicationPageChunk(
+  applicationCache: ApplicationCacheType,
+  applicationPageChunks: ApplicationPageChunk[],
+  remoteCacheKey: string
+): ApplicationPageChunk {
+  // if no cluster name chucks left, create a new array of chunks
+  if (applicationPageChunks.length === 0) {
+    // get all apps
+    let applications: ITransformedResource[] = []
+    if (applicationCache[remoteCacheKey].resources) {
+      applications = applicationCache[remoteCacheKey].resources
+    } else if (applicationCache[remoteCacheKey].resourceMap) {
+      applications = Object.values(applicationCache[remoteCacheKey].resourceMap).flat()
+    }
+    if (applications.length) {
+      // create array of frequency of name prefixes
+      const a = 'a'.charCodeAt(0)
+      const z = '0'.charCodeAt(0)
```

Comment:
Nit: If `a` is the letter 'a'... then I expect `z` to be the letter 'z', not zero. Can you choose better names, please?